File: Python/Programmers/Lv2/소수_만들기.py
def isPrime(num):
  for i in range(3,num):
    if num%i==0:
      return False
  return True

def solution(nums):
  size = len(nums)
  ans = 0
  for i in range(size-2):
    for j in range(i+1,size-1):
      for k in range(j+1,size):
        target = nums[i]+nums[j]+nums[k]
        if isPrime(target):
          ans+=1
  return ans

t = [1,2,3,4]
from itertools import combinations as cb
for i in cb(t,3):
  print(i)

a = solution(t)
# print(a)

File: AlgorithmTest/Programmers/src/lv2/소수_만들기.java
package lv2;


public class 소수_만들기 {

    public static void main(String[] args) {

    }
    public static int solution(int[] num){
        int size = num.length;
        int ans = 0;
        for (int i = 0; i < size-2; i++) {
            for (int j = i+1; j < size-1 ; j++) {
                for (int k = j+1; k < size; k++) {
                    int t = num[i]+num[j]+num[k];
                    if(isPrime(t)){
                        ans++;
                    }
                }
            }
        }
        return ans;

    }
    public static boolean isPrime(int number){
        for (int i = 3; i < number ; i++) {
            if(number%i==0){
                return false;
            }
        }
        return true;
    }
}

